annotation/util.scm

```diff
@@ -429,3 +432,28 @@
         ((and (cog-link? x) (null? (cog-outgoing-set x))) '())
         ((pair? x) (append (flatten (car x)) (flatten (cdr x))))
         (else (list x))))
+
+(define (do-find-organism gene)
+"
+  finds the organism(from human or virus etc ..) of a node
+  node can be GeneNode or MoleculeNode
+"
+  (run-query 
+    (Bind
+      (VariableList
+				(TypedVariable (Variable "$org") (Type 'ConceptNode))
+				(TypedVariable (Variable "$name") (Type 'ConceptNode)))
+      (AndLink
+        (Evaluation (Predicate "from_organism") (List gene (Variable "$org")))
+        (Evaluation (Predicate "has_name") (List (Variable "$org") (VariableNode "$name")))
+      )
+      (list
+        (Evaluation (Predicate "from_organism") (List gene (Variable "$org")))
+        (Evaluation (Predicate "has_name") (List (Variable "$org") (VariableNode "$name"))))
+  ))
+)
+; Cache results of do-find-organism for performance.
+(define memoize-find-organism (memoize-function-call do-find-organism)) 
```

Comment:
Minor comment. Can you rename this to `cache-find-organism` to be consistent with other function names?